04.ls/ls.rb

```diff
@@ -1,19 +1,114 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'ffi-xattr'
+require 'etc'
+
 COLUMN_COUNT = 3
 SPACE = 1
+PERMISSION_TYPE = ['--x', '-w-', '-wx', 'r--', 'r-x', 'rw-', 'rwx'].freeze
 
 def main
   options = {}
   opt = OptionParser.new
+  opt.on('-r') { |v| v }
```

Comment:
```suggestion
  opt.on('-r')
```

ブロックの内部で何もやってないので、これでも動きそうです